orttraining/orttraining/python/training/ortmodule/ort_triton/_utils.py

```diff
@@ -0,0 +1,172 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+
+import re
+import uuid
+from collections import defaultdict
+from typing import Any, List, Tuple
+
+import numpy as np
+from onnx import GraphProto, NodeProto, TensorProto, helper, numpy_helper
+
+
+def gen_unique_name(prefix: str) -> str:
+    return prefix + "_" + uuid.uuid4().hex[:8]
+
+
+def _topological_soft_internal(node, visited, output_consumers, sorted_nodes):
+    visited.add(node.name)
+    for next_node in output_consumers[node.name]:
+        if next_node.name not in visited:
```

Comment:
Should we have a pre-check to assure names of each node are unique?
If there is the case, which might be super hard to debug.

Reply:
For training this is not that possible to happen as the graph is exported online from torch, and the graph proceseed here is only sub-graphs of the graph.